### ISSUE TITLE
Docs: local plugin folder name is not clear

### ISSUE THREAD

**Issue by borekb** [gatsbyjs/gatsby#8261]
I was following the [Local plugins section](https://github.com/gatsbyjs/gatsby/blob/ba2ea3871bdef6f822c9a9d46ecb98ad796dfd50/docs/docs/plugin-authoring.md#local-plugins) of the docs and it was not clear that the folder name is significant for the plugin to be discovered – I thought that the `name` field in `package.json` is used.

For example, this doesn't work:

```
plugins
└── my-own-plugin
    └── package.json
```

package.json:

```
{
  "name": "gatsby-source-example"
}
```

gatsby-config.js:

```
module.exports = {
  plugins: [
    'gatsby-source-example',
  ]
}
```

This leads to

> Error: Unable to find plugin "gatsby-source-example". Perhaps you need to install its package?

The correct way to load the plugin is:

```
module.exports = {
  plugins: [
    'my-own-plugin',
  ]
}
```

**Comment by kakadiadarpan** [comment]
Hi @borekb, would you like to raise a PR to fix the [`local plugins`](https://github.com/gatsbyjs/gatsby/blob/master/docs/docs/plugin-authoring.md) documentation?

**Comment by thomasrumbold** [comment]
@borekb , @kakadiadarpan, I've made a minor update here that may do the trick: https://github.com/gatsbyjs/gatsby/pull/8275.

**Comment by borekb** [comment]
@thomasrumbold beat me to it :) Before submitting a PR, I wanted to verify that the current behavior is correct – I'd expect the `name` field in `package.json` be the package's name, not its folder name.

Also this line could be considered for revision in that PR: https://github.com/gatsbyjs/gatsby/blame/e4fb8e4aee271e6e3ce9044811e122e6d7c76645/docs/docs/plugin-authoring.md#L39

In any case, a docs update will be welcome, thanks @thomasrumbold!

**Comment by mikelax** [comment]
Closed as #8275 addresses this.
